src/Apps/Sell/Components/BottomFormNavigation.tsx

```diff
@@ -75,55 +77,89 @@ const BottomFormNextButton = () => {
   const { isLoggedIn } = useSystemContext()
   const { showAuthDialog } = useAuthDialog()
   const { trackTappedContinueSubmission } = useSubmissionTracking()
-
   const {
-    state: { loading },
-  } = useSellFlowContext()
+    submitMutation: associateSubmissionMutation,
+  } = useAssociateSubmission()
+  const { value, clearValue } = useAuthIntent()
 
   const {
-    actions,
-    state: { submission, isSubmitStep, nextStep },
+    actions: { goToNextStep },
+    state: { submission, isSubmitStep, nextStep, loading },
   } = useSellFlowContext()
 
-  const onNext = async () => {
-    if (!isLoggedIn) {
+  const onContinue = async () => {
+    trackTappedContinueSubmission(submission?.internalID, nextStep)
+
+    if (!isLoggedIn && isSubmitStep) {
+      await submitForm()
+
       showAuthDialog({
-        mode: "Login",
+        mode: "SignUp",
         options: {
-          title: () => {
-            return "Log in to submit an artwork for sale"
+          title: mode =>
+            mode === "Login"
+              ? "Log in to complete your submission"
+              : "Sign up to complete your submission",
+          afterAuthAction: {
+            action: "submitSubmission",
+            kind: "submission",
+            objectId: submission?.externalId as string,
           },
         },
         analytics: {
-          contextModule: ContextModule.consignSubmissionFlow,
-          intent: Intent.login,
+          contextModule: ContextModule.sell,
+          intent: Intent.consign,
+          trigger: "click",
         },
       })
 
       return
+    } else {
+      await submitAndNavigateToNextStep()
     }
+  }
 
+  const submitAndNavigateToNextStep = useCallback(async () => {
     setIsSubmitting(true)
 
-    trackTappedContinueSubmission(submission?.internalID, nextStep)
-
     try {
       await submitForm()
 
-      await actions.goToNextStep()
+      await goToNextStep()
     } catch (error) {
       logger.error("Error submitting form", error)
     } finally {
       setIsSubmitting(false)
     }
-  }
+  }, [submitForm, goToNextStep])
+
+  const onAfterAuthCallback = useCallback(async () => {
+    try {
+      setIsSubmitting(true)
+      await associateSubmissionMutation({
+        variables: { input: { id: submission?.externalId as string } },
+      })
+      await submitAndNavigateToNextStep()
+    } catch (error) {
+      logger.error("Something went wrong.", error)
+      setIsSubmitting(false)
+    }
```

Comment:
Would it make sense to set `isSubmitting` to `false` in a `finally` block? Right now it isn't relevant because the `thank-you` step does not display any loading state. But it would probably be cleaner.